package/MDAnalysis/core/topologyattrs.py

```diff
@@ -2250,15 +2250,42 @@ def _bondDict(self):
     def set_atoms(self, ag):
         return NotImplementedError("Cannot set bond information")
 
-    def get_atoms(self, ag):
+    def get_atoms(self, ag, outside=True):
+        """
+        Get subset for atoms.
+
+        Parameters
+        ----------
+        ag : AtomGroup
+        outside : bool (optional)
+            Whether to include connections to atoms outside the given
+            AtomGroup.
+
+        .. versionchanged:: 1.1.0
+            Added the ``outside`` keyword. Set to ``True`` by default
+            to give the same behavior as previously
+        """
         try:
             unique_bonds = set(itertools.chain(
                 *[self._bondDict[a] for a in ag.ix]))
         except TypeError:
             # maybe we got passed an Atom
             unique_bonds = self._bondDict[ag.ix]
-        bond_idx, types, guessed, order = np.hsplit(
-            np.array(sorted(unique_bonds), dtype=object), 4)
+        unique_bonds = np.array(sorted(unique_bonds), dtype=object)
+        if not outside:
```

Comment:
Why not 
```
if outside: 
    <deprecation code>
else:
    <np.all>
```
It makes it more readable imo

Reply:
I started with ``strict=False`` 😅

This will make it easier to add to 2.0, though, where we will not have the deprecation warning.